src/lib.rs

```diff
@@ -179,20 +192,55 @@ impl Catalog {
 }
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-struct Message {
-    id: String,
-    context: Option<String>,
-    translated: Vec<String>,
+/// `Message` represents a message that can be translated. It contains
+/// the original string (ID), an optional plural form, an optional context
+/// for disambiguation, and the translated strings for the message.
+pub struct Message {
+    /// The original string to be translated, used as the key for looking up
+    /// translations.
+    pub id: String,
+    /// An optional context for the translation, used for disambiguation
+    /// when the same original string can have different translations
+    /// depending on its usage.
+    pub context: Option<String>,
+    /// Translated strings for the message. Contains one string for each
+    /// plural form in the target language.
+    pub translated: Vec<String>,
+	/// An optional plural form of the original string, used for languages
+    /// that have more than one form for plurals.
+    pub plural: Option<String>
 }
 
 impl Message {
-    fn new<T: Into<String>>(id: T, context: Option<T>, translated: Vec<T>) -> Self {
+	/// Constructs a new `Message` instance with the given id, context and translated strings.
+    fn new<T: Into<String>>(
+        id: T,
+        context: Option<T>,
+        translated: Vec<T>,
+    ) -> Self {
```

Comment:
We should revert the pure formatting changes here to avoid the noise.

Reply:
@rmccue This has been resolved.